src/theory/quantifiers/sygus/rcons_obligation_info.h

```diff
@@ -0,0 +1,108 @@
+/*********************                                                        */
+/*! \file rcons_obligation_info.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Abdalrhman Mohamed
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief utility class for Sygus Reconstruct module
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__QUANTIFIERS__RCONS_OBLIGATION_INFO_H
+#define CVC4__THEORY__QUANTIFIERS__RCONS_OBLIGATION_INFO_H
+
+#include "expr/node.h"
+
+namespace CVC4 {
+namespace theory {
+namespace quantifiers {
+
+/**
+ * A utility class for Sygus Reconstruct obligations. An obligation is a
+ * builtin term t and a SyGuS type T, and indicates that we are trying to build
+ * a term of type T whose builtin analog is equivalent to t.
+ */
+class RConsObligationInfo
+{
+ public:
+  /**
+   * Constructor. Default value needed for maps.
+   *
+   * @param builtin builtin term to reconstruct
+   */
+  explicit RConsObligationInfo(const Node& builtin = Node::null());
```

Comment:
You should always use `TNode` instead of `const Node&` I believe.

Reply:
I wanted to use `const Node&` because `builtin` is assigned to `d_builtin` and I wanted to avoid a conversion back and forth. Not worth the effort though. Changed everything to `Node`.